internal/rules/migrate.go

```diff
@@ -16,6 +18,64 @@ type Migrate struct {
 	Symlink bool
 }
 
+func MoveDirectory(source, dest string) error {
+	err := copy.Copy(source, dest)
+	if err != nil {
+		return err
+	}
+
+	err = os.RemoveAll(source)
+	if err != nil {
+		tui.Warn("Failed to remove original directory: %s", err)
+	}
+
+	return nil
+}
+
+// Adapted from https://gist.github.com/var23rav/23ae5d0d4d830aff886c3c970b8f6c6b
+func MoveFile(source, dest string) error {
+	inputFile, err := os.Open(source)
+	if err != nil {
+		return err
+	}
+
+	outputFile, err := os.Create(dest)
+	if err != nil {
+		inputFile.Close()
+		return err
+	}
+	defer outputFile.Close()
+
+	_, err = io.Copy(outputFile, inputFile)
+	inputFile.Close()
+	if err != nil {
+		return err
+	}
+
+	// The copy was successful, so now delete the original file
+	err = os.Remove(source)
+	if err != nil {
+		tui.Warn("Failed to remove original file: %s", err)
+	}
+
+	return nil
+}
+
+// Try to move file/directory with os.Rename and if that fails, do a copy + delete
+func Move(source, dest string) error {
+	err := os.Rename(source, dest)
+	if err == nil {
+		return nil
+	}
+
+	fi, err := os.Stat(source)
+	if fi.Mode().IsDir() {
+		return MoveDirectory(source, dest)
+	}
+
+	return MoveFile(source, dest)
+}
+
```

Comment:
Could you move this logic to `internal/utils/files.go` and see if `MoveFile` needs to be changed or replaced?

Reply:
Good idea. I didn't notice that there was already a ``MoveFile`` function.